Demo/Common/IntegrationExamples/Protocols/AdvancedFlowExampleProtocol.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+import AdyenNetworking
+import AdyenSession
+
+internal protocol AdvancedFlowExampleProtocol: AnyObject, APIClientAware {
+    var context: AdyenContext { get }
+    var palApiClient: APIClientProtocol { get }
+    func requestPaymentMethods(order: PartialPaymentOrder?,
+                                        completion: ((PaymentMethods?, Error?) -> Void)?)
+}
+
+extension AdvancedFlowExampleProtocol {
+
+    internal var context: AdyenContext {
+        var analyticsConfiguration = AnalyticsConfiguration()
+        analyticsConfiguration.isEnabled = true
+        return AdyenContext(apiContext: ConfigurationConstants.apiContext,
+                            payment: ConfigurationConstants.current.payment,
+                            analyticsConfiguration: analyticsConfiguration)
+    }
+
+    internal var palApiClient: APIClientProtocol {
+        let context = DemoAPIContext(environment: ConfigurationConstants.classicAPIEnvironment)
+        return DefaultAPIClient(apiContext: context)
+    }
+
+    internal func requestPaymentMethods(order: PartialPaymentOrder?,
+                                        completion: ((PaymentMethods?, Error?) -> Void)?) {
```

Comment:
Maybe use `Result<PaymentMethods,Error>` instead?

Reply:
I need to use optional error in the completion and it does not accept an optional error. So, I cannot use this.